nomos-cli/src/cmds/chat/mod.rs

```diff
@@ -0,0 +1,260 @@
+        let network = serde_yaml::from_reader::<
+            _,
+            <NetworkService<Libp2p> as ServiceData>::Settings,
+        >(std::fs::File::open(&self.network_config)?)?;
+        let da_protocol = self.da_protocol.clone();
+        // setup terminal
+        enable_raw_mode()?;
+        let mut stdout = io::stdout();
+        execute!(stdout, EnterAlternateScreen, EnableMouseCapture)?;
+        let backend = CrosstermBackend::new(stdout);
+        let mut terminal = Terminal::new(backend)?;
+
+        let node_addr = Some(self.node.clone());
+
+        let (payload_sender, payload_receiver) = tokio::sync::mpsc::unbounded_channel();
+        let (status_sender, status_updates) = std::sync::mpsc::channel();
+
+        std::thread::spawn(move || {
+            OverwatchRunner::<DisseminateApp>::run(
+                DisseminateAppServiceSettings {
+                    network,
+                    send_blob: Settings {
+                        payload: Arc::new(Mutex::new(payload_receiver)),
+                        timeout: DEFAULT_TIMEOUT,
+                        da_protocol,
+                        status_updates: status_sender,
+                        node_addr,
+                        output: None,
+                    },
+                },
+                None,
+            )
+            .unwrap()
+            .wait_finished()
+        });
+
+        let app = App {
+            input: Input::default(),
+            username: None,
+            messages: Vec::new(),
+            message_status: None,
+            message_in_flight: false,
+            last_updated: Instant::now(),
+            payload_sender,
+            status_updates,
+            node: self.node.clone(),
+        };
+
+        run_app(&mut terminal, app);
+
+        // restore terminal
+        disable_raw_mode()?;
+        execute!(
+            terminal.backend_mut(),
+            LeaveAlternateScreen,
+            DisableMouseCapture
+        )?;
+        terminal.show_cursor()?;
+
+        Ok(())
+    }
+}
+
+fn run_app<B: Backend>(terminal: &mut Terminal<B>, mut app: App) {
+    let (message_tx, message_rx) = std::sync::mpsc::channel();
+    let node = app.node.clone();
+    std::thread::spawn(move || check_for_messages(message_tx, node));
+    loop {
+        terminal.draw(|f| ui::ui(f, &app)).unwrap();
+
+        if let Ok(update) = app.status_updates.try_recv() {
+            if let Status::Done = update {
+                app.message_in_flight = false;
+                app.message_status = None;
+            }
+            app.message_status = Some(update);
+            app.last_updated = Instant::now();
+        }
+
+        if let Ok(messages) = message_rx.try_recv() {
+            app.messages.extend(messages);
+        }
+
+        // Do not block rendering if there's no user input available
+        if !event::poll(Duration::from_millis(100)).unwrap() {
+            continue;
+        }
```

Comment:
![image](https://github.com/logos-co/nomos-node/assets/5462944/c483f81f-cdbc-4cd7-83b9-228d65e8d2cc)
I've seen the message status wes stuck at `WaitingAttestations`. I'm taking a look at it deeply, but have you seen this symptom before?

Reply:
Ah, never mind. I misconfigured it.